shared/gh/js/utils/gh.utils.templates.js

```diff
@@ -110,34 +134,44 @@ define(['exports', 'gh.constants'], function(exports, constants) {
     /**
      * Render a template and either return the HTML or populate a target container with the result
      *
-     * @param  {Element|String}    $template    jQuery element representing the HTML element that contains the template or jQuery selector for the template container
-     * @param  {Object}            [data]       JSON object representing the values used to process the template
-     * @param  {Element|String}    [$target]    jQuery element representing the HTML element in which the template output should be put, or jQuery selector for the output container
-     * @return {String}                         The rendered HTML
-     * @throws {Error}                          Error thrown when no template has been provided
+     * @param  {String}            partial                    The name of the partial to render
+     * @param  {Object}            [data]                     JSON object representing the values used to process the template
+     * @param  {Element|String}    [$target]                  jQuery element representing the HTML element in which the template output should be put, or jQuery selector for the output container
+     * @param  {Function}          [callback]                 Optional callback to execute after the partial has been retrieved and rendered
+     * @param  {String}            [callback.renderedHTML]    The rendered HTML String
+     * @return {String}                                       The rendered HTML String
+     * @throws {Error}                                        Error thrown when no template has been provided
      */
-    var renderTemplate = exports.renderTemplate = function($template, data, $target) {
-        if (!$template) {
-            throw new Error('No valid template has been provided');
+    var renderTemplate = exports.renderTemplate = function(partial, data, $target, callback) {
+        if (!_.isString(partial)) {
+            throw new Error('No valid partial name has been provided');
         }
 
```

Comment:
Can you add some parameter validation for `data` and `$target` as well here? In case they are wrongfully provided...

```
} else if (data && !_.isObject(data)) {
    ...
} else if ($target && (!_.isObject($target) && !_.isString($target))) {
    ...
}
```